File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int asc=1;
        int dsc=1;
        int ans=1;
        for(int i=0 ; i<nums.size()-1;i++){
            if(nums[i+1]>nums[i]){
                asc++;
                dsc=1;
            }
            else if(nums[i+1]<nums[i]){
                dsc++;
                asc=1;
            }
            else{
                dsc=1;
                asc=1;
            }
            ans=max(ans,max(asc,dsc));
        }
        return ans;
        
    }
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        asc=0
        dsc=0
        ans=0
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1] :
                dsc=0
                asc=asc+1
            elif nums[i]<nums[i-1]:
                asc=0
                dsc+=1
            else :
                asc=0
                dsc=0
            ans=max(ans,max(asc,dsc))
        return ans+1
                
        